lib/beaker/dsl/helpers.rb

```diff
@@ -553,7 +553,7 @@ def with_puppet_running_on host, conf_opts, testdir = host.tmpdir(File.basename(
         end
 
         begin
-          backup_file = backup_the_file(host, host['puppetconfdir'], testdir, 'puppet.conf')
+          backup_file = backup_the_file(host, host.puppet['confdir'], testdir, 'puppet.conf')
```

Comment:
I believe `with_puppet_running_on` is only ever called on the master node, in which case, I think we always want to do `host.puppet('master')['confdir']`.

Currently the agent and master share the confdir (so it shouldn't matter), but the confdir may be different in the future, e.g. the master's vardir is changing https://tickets.puppetlabs.com/browse/SERVER-369